src/main/java/org/jenkins/tools/test/model/PluginRemoting.java

```diff
@@ -73,71 +72,34 @@ public PluginRemoting(String hpiRemoteUrl){
     public PluginRemoting(File pomFile){
         this.pomFile = pomFile;
     }
-	
-    private String retrievePomContent() throws PluginSourcesUnavailableException{
-        if (hpiRemoteUrl != null) {
-            return retrievePomContentFromHpi();
-        } else {
-            return retrievePomContentFromXmlFile();
-        }
-    }
 
-    private String retrievePomContentFromHpi() throws PluginSourcesUnavailableException {
+    private String retrievePomContent() {
         try {
-	    return retrievePomContentFromHpiRemoteUrl(new URL(hpiRemoteUrl));
-	} catch (MalformedURLException e) {
-	    LOGGER.log(Level.WARNING, "HPI reference {0} is not a remote URL", hpiRemoteUrl);
-	    return retrievePomContentFromHpiFileReference();
-	}
-    }
-
-    private String retrievePomContentFromHpiRemoteUrl(URL url) throws PluginSourcesUnavailableException {
-    	try {
-    	    return retrievePomContentFromInputStream(url.openStream());
-	} catch (IOException e) {
-	    LOGGER.log(Level.WARNING, "Failed to retrieve POM content from HPI remote URL", e);
-          throw new PluginSourcesUnavailableException("Failed to retrieve POM content from HPI remote URL", e);
-	}
-    }
-
-    private String retrievePomContentFromHpiFileReference() throws PluginSourcesUnavailableException {
-    	try {
-	    String fileReference = hpiRemoteUrl.replaceAll("jar:", "").replaceAll("!/name.hpi", "");
-    	    return retrievePomContentFromInputStream(FileUtils.openInputStream(new File(fileReference)));
-	} catch (IOException e) {
-	    LOGGER.log(Level.WARNING, "Failed to retrieve POM content from HPI file reference", e);
-          throw new PluginSourcesUnavailableException("Failed to retrieve POM content from HPI file reference", e);
-	}
+            if (hpiRemoteUrl != null) {
+                return retrievePomContentFromHpi();
+            } else {
+                return retrievePomContentFromXmlFile();
+            }
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
     }
 
-    private String retrievePomContentFromInputStream(InputStream pluginUrlStream) throws PluginSourcesUnavailableException {
-    	try (ZipInputStream zin = new ZipInputStream(pluginUrlStream)) {
-            ZipEntry zipEntry = zin.getNextEntry();
-            while(!zipEntry.getName().startsWith("META-INF/maven") || !zipEntry.getName().endsWith("pom.xml")){
-                zin.closeEntry();
-                zipEntry = zin.getNextEntry();
+    private String retrievePomContentFromHpi() throws IOException {
+        URL url = new URL(hpiRemoteUrl);
+        try (InputStream is = url.openStream(); ZipInputStream zis = new ZipInputStream(is)) {
+            ZipEntry ze;
+            while ((ze = zis.getNextEntry()) != null) {
+                if (ze.getName().startsWith("META-INF/maven") && ze.getName().endsWith("pom.xml")) {
+                    return new String(zis.readAllBytes(), StandardCharsets.UTF_8);
+                }
```

Comment:
this appears to be missing a call to [`ZipInputStream.closeEntry()`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/zip/ZipInputStream.html#closeEntry()) in order to re-position the pointer for the next read.

The javadoc is super unhelpful if you actually need to call this or not - but given the absence of direction to say either way - it would be prudent to be defensive against any changes here.

```suggestion
                    zis.closeEntry();
                }
```

Reply:
scratch that comment - the [javadoc](https://github.com/openjdk/jdk/blob/bc750f70f2ac1f14f5b8e6236c593dcbe99cb12f/src/java.base/share/classes/java/util/zip/ZipInputStream.java#L48-L65) has been amended (post java17) and the [example shows](https://github.com/openjdk/jdk/blob/bc750f70f2ac1f14f5b8e6236c593dcbe99cb12f/src/java.base/share/classes/java/util/zip/ZipInputStream.java#L48-L65) no need to perform a close and the code inspection shows that [closeEntry is automatically called](https://github.com/openjdk/jdk/blob/bc750f70f2ac1f14f5b8e6236c593dcbe99cb12f/src/java.base/share/classes/java/util/zip/ZipInputStream.java#L149).